### PR TITLE
Update to latest dependencies and merge config CLI plumbing changes

### PR DESCRIPTION
**Pull request by dcoutts**
This takes most of #79 and #89 but hopefully selected in a more careful and comprehensible way.

It updates the deps on consensus, network, ledger etc to the latest. The `cardano-node` built from here does sync with the `cardano-byron-proxy` from the `avieth/chaindb` branch.

TODOs after this:
 * Not yet checked the local tx submission
 * Not yet checked the node client example
 * Add the scripts and example config files frin #79 and/or #89 
 * Check the README instructions to make sure they're in sync with reality

### CODE REVIEW

File: app/Run.hs [hunk 1]
```diff
@@ -178,7 +179,7 @@ runNode nodeCli@NodeCLIArguments{..} loggingLayer cc = do
 -- | Sets up a simple node, which will run the chain sync protocol and block
 -- fetch protocol, and, if core, will also look at the mempool when trying to
 -- create a new block.
-handleSimpleNode :: forall blk. (RunNode blk, TraceConstraints blk)
+handleSimpleNode :: forall blk. (RunNode blk, TraceConstraints blk, Show (GenTxId blk))
```
Comment (mrBliss) [inline app/Run.hs]: There is no need for this change as `Show (GenTxId blk)` is part of `TraceConstraints`.

Comment (dcoutts) [reply app/Run.hs]: Ah yes, it is now after we rebased on the other PR. Thanks.

**Comment by dcoutts** [comment]
bors ping

**Pull request merged**